### PR TITLE
Some optimizations for traces with 100s of shaders/textures.

### PR DESCRIPTION
**Pull request by pmuetschard**
- Don't starve async scheduled runnables, if repaint is slow
- Only update the shader dropdown if required to do so
- Only load texture thumbnails that are visible
- Batch updates to the texture table

### CODE REVIEW

File: gapic/src/main/com/google/gapid/widgets/VisibilityTrackingTreeViewer.java [hunk 1]
```diff
@@ -70,6 +70,13 @@ private void updateVisibility() {
   }
 
   @Override
+  protected void inputChanged(Object input, Object oldInput) {
+    // Cause all items to receive a show event.
```
Comment (ben-clayton) [inline gapic/src/main/com/google/gapid/widgets/VisibilityTrackingTreeViewer.java]: all new visible items, right?

File: gapis/resolve/resource_data.go [hunk 2]
```diff
@@ -74,9 +74,10 @@ func buildResources(ctx context.Context, p *path.Command) (*ResolvedResources, e
 	for k, v := range resources {
 		res, err := v.ResourceData(ctx, state)
 		if err != nil {
-			return nil, fmt.Errorf("Could not get data for resource %v", k)
+			resourceData[k] = err;
```
Comment (ben-clayton) [inline gapis/resolve/resource_data.go]: [Okay, you are making me feel uneasy now.](https://github.com/google/gapid/pull/212#discussion_r109881143)
I swear I didn't see this before making those comments...

**Comment by AWoloszyn** [comment]
This greatly helps for the enormous traces I have been looking at.

File: gapic/src/main/com/google/gapid/widgets/VisibilityTrackingTreeViewer.java (hunk 1 above)
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/widgets/VisibilityTrackingTreeViewer.java]: Yes. Only the visible ones. Updated the comment.

File: gapis/resolve/resource_data.go (hunk 2 above)
Comment (pmuetschard) [inline gapis/resolve/resource_data.go]: Oh, and here I was worried that isErr is not :notes:idiomatic go:notes:

**Comment by ben-clayton** [comment]
> Batch updates to the texture table

Where is this batching of which you speak?

**Comment by pmuetschard** [comment]
> Where is this batching of which you speak?

https://github.com/google/gapid/pull/210/files#diff-a48d43384608915e3d14df3b796da55bR537

"intercepts" .refresh() calls, and bundles those that come in quick succession together as one.

**Comment by ben-clayton** [comment]
Ack. Thanks.

**Pull request merged**